agb/src/display/font.rs

```diff
@@ -84,43 +74,64 @@ pub struct TextRenderer<'a> {
     current_y_pos: i32,
     font: &'a Font,
     tile_pos: Vector2D<u16>,
-    vram_manager: &'a mut VRamManager,
-    bg: &'a mut RegularMap,
-    background_colour: u8,
-    foreground_colour: u8,
     tiles: HashMap<(i32, i32), DynamicTile<'a>>,
 }
 
-impl<'a> Write for TextRenderer<'a> {
+pub struct TextWriter<'a, 'b: 'a, 'c> {
```

Comment:
My only comment about this API would be the use of 3 lifetime parameters. Playing around locally, I could reduce this to 2 by getting rid of `'b` entirely, and the example still compiled.

I'd prefer minimising lifetime parameters because it can make the API more complicated then in needs to be :). We can always put them in if needed.

Reply:
Thanks,  would you mind being more specific with regards to this change? If I try to remove the `'b` lifetime param and replace it on the `vram_manager` with `'a` or `'b` it gives me borrow checking/lifetime errors.

Reply:
I've done it here :) (and also fixed the remaining example)

https://github.com/jembishop/agb/compare/textrender-rework...gwilymk:gba:textrender-rework-gwilym